From 45d0d8f11c1fe1f68072617f1341dd5013877244 Mon Sep 17 00:00:00 2001
From: Huicong Xu <xhc@rock-chips.com>
Date: Thu, 1 Mar 2018 16:52:13 +0800
Subject: [PATCH] phy: rockchip: inno-hdmi: fix hdmi can't display after change
 mode

for rk322x power down post-PLL must to both set rege0[5]=0 and set
pre pll pre-PLL unlock. So power down pre-PLL before post-PLL power
down

Change-Id: If0eb325b10bb6eb117b0a61d5852e9aae9d92ba6
Signed-off-by: Huicong Xu <xhc@rock-chips.com>
---
 drivers/phy/rockchip/phy-rockchip-inno-hdmi-phy.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/phy/rockchip/phy-rockchip-inno-hdmi-phy.c b/drivers/phy/rockchip/phy-rockchip-inno-hdmi-phy.c
index c4af0ebe689f..ca494b116107 100644
--- a/drivers/phy/rockchip/phy-rockchip-inno-hdmi-phy.c
+++ b/drivers/phy/rockchip/phy-rockchip-inno-hdmi-phy.c
@@ -602,6 +602,7 @@ inno_hdmi_phy_rk3228_power_on(struct inno_hdmi_phy *inno,
 	inno_update_bits(inno, 0x02, PDATAEN_MASK, PDATAEN_DISABLE);
 
 	/* Power down Post-PLL */
+	inno_update_bits(inno, 0xe0, PRE_PLL_POWER_MASK, PRE_PLL_POWER_DOWN);
 	inno_update_bits(inno, 0xe0, POST_PLL_POWER_MASK, POST_PLL_POWER_DOWN);
 
 	/* Post-PLL update */
@@ -635,6 +636,7 @@ inno_hdmi_phy_rk3228_power_on(struct inno_hdmi_phy *inno,
 
 	/* Power up Post-PLL */
 	inno_update_bits(inno, 0xe0, POST_PLL_POWER_MASK, POST_PLL_POWER_UP);
+	inno_update_bits(inno, 0xe0, PRE_PLL_POWER_MASK, PRE_PLL_POWER_UP);
 
 	/* BandGap enable */
 	inno_update_bits(inno, 0xe1, BANDGAP_MASK, BANDGAP_ENABLE);
-- 
2.35.3

